my-page-app/src/components/admin/Admin.jsx

```diff
@@ -1,129 +1,165 @@
+                    if (!budgetTypeListContains(extractedBudgetTypes, budget)) {
+                        budget.budgetType.balanceIsHours = false;
+                        extractedBudgetTypes.push(budget.budgetType);
+                        if (budget.budgetType.allowTimeBalance) {
+                            let budgetTypeHours = Object.assign({}, budget.budgetType);
+                            budgetTypeHours.balanceIsHours = true;
+                            budgetTypeHours.name += "(timer)";
+                            extractedBudgetTypes.push(budgetTypeHours);
+                        }
+                    }
+                }
+            ))
+            setBudgetTypes(extractedBudgetTypes);
         }
-      ))
-      setBudgetTypes(extractedBudgetTypes);
-    }
-  };
+    };
 
-  const budgetTypeListContains = (extractedBudgetTypes, newBudget) => {
-    return extractedBudgetTypes.some( (budgetType) => (budgetType.id === newBudget.budgetType.id) )
-  };
+    const budgetTypeListContains = (extractedBudgetTypes, newBudget) => {
+        return extractedBudgetTypes.some((budgetType) => (budgetType.id === newBudget.budgetType.id))
+    };
 
-  const getBudgetBalanceForType = (budgets, type) => {
-    var foundBudget = budgets.find(budget => budget.budgetType.id === type.id);
-    if (type.balanceIsHours) {
-      return budgetBalanceHours(foundBudget);
-    } else {
-      return budgetBalance(foundBudget);
-    }
-  };
+    const getBudgetBalanceForType = (budgets, type) => {
+        var foundBudget = budgets.find(budget => budget.budgetType.id === type.id);
+        if (type.balanceIsHours) {
+            return budgetBalanceHours(foundBudget);
+        } else {
+            return budgetBalance(foundBudget);
+        }
+    };
 
-  const budgetBalance = (budget) => {
-    if (budget) {      
-      return budget.balance.toLocaleString('no-NO', { maximumFractionDigits: 2, style: 'currency', currency: 'NOK' });
-    } else {
-      return "-"
-    }
-  };
+    const budgetBalance = (budget) => {
+        if (budget) {
+            return budget.balance.toLocaleString('no-NO', {
+                maximumFractionDigits: 2,
+                style: 'currency',
+                currency: 'NOK'
+            });
+        } else {
+            return "-"
+        }
+    };
 
-  const budgetBalanceHours = (budget) => {
-    if (budget) {
-      return budget.sumHours + (budget.sumHours === 1 ? " time" : " timer");
+    const budgetBalanceHours = (budget) => {
+        if (budget) {
+            return budget.sumHours + (budget.sumHours === 1 ? " time" : " timer");
+        } else {
+            return "-"
+        }
+    };
+
+    const handleExpandUser = (user, event) => {
+        // Check if the clicked target is not a child of the expanded area
+        const expandedArea = document.querySelector(`[data-expanded-user="${user.email}"]`);
+        if (expandedArea == null || !expandedArea.contains(event.target)) {
```

Comment:
```suggestion
        if (expandedArea === null || !expandedArea.contains(event.target)) {
```

Always prefer triple equals. This ensure the type equivalence as well.